tests/std/tests/P0608R3_sane_variant_converting_constructor/test.cpp

```diff
@@ -0,0 +1,174 @@
+#include <variant>
+#include <vector>
+
+using namespace std;
+
+struct double_double {
+    double_double(double x) : x_(x) {}
+
+    double x_;
+};
+struct convertible_bool {
+    convertible_bool(bool x) : x_(x) {}
+    ~convertible_bool() = default;
+
+    operator bool() const noexcept {
+        return x_;
+    }
+
+    bool x_;
+};
+
+// P0608R3 examples
+static_assert(is_constructible_v<variant<string, bool>, const char*>);
+static_assert(is_constructible_v<variant<string, bool>, string>);
+static_assert(is_constructible_v<variant<char, optional<char16_t>>, char16_t>);
+static_assert(is_constructible_v<variant<int, reference_wrapper<double>>, double&>);
+static_assert(is_constructible_v<variant<float, int>, char>);
+#ifndef __EDG__ // TRANSITION, ...
+static_assert(is_constructible_v<variant<float, long int>, int>);
+static_assert(is_constructible_v<variant<float, long long int>, int>);
+static_assert(is_constructible_v<variant<float, long, double>, int>);
+static_assert(is_constructible_v<variant<float, vector<int>, long long int>, int>);
+#endif // !__EDG__
+static_assert(is_constructible_v<variant<float, int, long long int>, char>);
+
+#ifndef __EDG__ // TRANSITION, ...
+static_assert(!is_constructible_v<variant<float>, int>);
+static_assert(!is_constructible_v<variant<float, vector<int>>, int>);
+#endif // !__EDG__
+static_assert(!is_constructible_v<variant<float, char>, int>);
+
+// P1957R2 examples
+static_assert(is_constructible_v<variant<bool, int>, bool>);
+static_assert(is_constructible_v<variant<bool, int>, std::bitset<4>::reference>);
+static_assert(is_constructible_v<variant<bool>, std::bitset<4>::reference>);
+
+// More examples
+static_assert(is_constructible_v<variant<double_double>, double>);
+static_assert(is_constructible_v<variant<vector<vector<int>>, optional<int>, int>, int>);
+static_assert(is_constructible_v<variant<vector<vector<int>>, optional<int>>, int>);
+static_assert(is_constructible_v<variant<vector<int>, optional<int>, float>, int>);
+static_assert(is_constructible_v<variant<bool>, convertible_bool>);
+static_assert(is_constructible_v<variant<bool, int>, convertible_bool>);
+static_assert(is_constructible_v<variant<convertible_bool>, bool>);
+static_assert(is_constructible_v<variant<float, bool, convertible_bool>, convertible_bool>);
+static_assert(is_constructible_v<variant<float, bool, convertible_bool>, bool>);
+static_assert(is_constructible_v<variant<char, int>, bool>);
+
+#ifndef __EDG__ // TRANSITION, ...
+#ifndef __clang__ // TRANSITION, ...
+static_assert(!is_constructible_v<variant<double_double>, int>);
+#endif // !__clang__
+static_assert(!is_constructible_v<variant<float>, unsigned int>);
+#endif // !__EDG__
+static_assert(!is_constructible_v<variant<float, long int, long long int>, int>);
+
+void test_variant_constructor_P0608R3() {
+    // P0608R3 runtime checks
+    variant<string, bool> a = "abc"; // string
+    assert(a.index() == 0);
+    assert(get<0>(a) == string("abc"));

```

Comment:
Constructing a temporary `string` here is unnecessary; we can directly compare to the string literal.